terraform/spec/dependabot/terraform/registry_client_spec.rb

```diff
@@ -33,73 +35,108 @@
     )
   end
 
-  let(:provider_dependency) do
-    Dependabot::Dependency.new(
-      name: "hashicorp/aws",
-      version: "0.9.3",
-      package_manager: "terraform",
-      previous_version: "3.19.0",
-      requirements: [{
-        requirement: "3.40.0",
-        groups: [],
-        file: "main.tf",
-        source: {
-          type: "provider",
-          registry_hostname: "registry.terraform.io",
-          module_identifier: "hashicorp/aws"
-        }
-      }],
-      previous_requirements: [{
-        requirement: "3.19.0",
-        groups: [],
-        file: "main.tf",
-        source: {
-          type: "provider",
-          registry_hostname: "registry.terraform.io",
-          module_identifier: "hashicorp/aws"
-        }
-      }]
-    )
-  end
-
   it "fetches provider versions", :vcr do
-    client = described_class.new(hostname: described_class::PUBLIC_HOSTNAME)
     response = client.all_provider_versions(identifier: "hashicorp/aws")
     expect(response.max).to eq(Gem::Version.new("3.40.0"))
   end
 
+  it "fetches provider versions from a custom registry" do
+    hostname = "registry.example.org"
+    stub_request(:get, "https://#{hostname}/v1/providers/hashicorp/aws/versions").and_return(
+      status: 200,
+      body: {
+        "id": "hashicorp/aws",
+        "versions": [{ "version": "3.42.0" }, { "version": "3.41.0" }]
+      }.to_json
+    )
+    client = described_class.new(hostname: hostname)
+    response = client.all_provider_versions(identifier: "hashicorp/aws")
+    expect(response).to match_array([
+      Gem::Version.new("3.42.0"),
+      Gem::Version.new("3.41.0")
+    ])
+  end
+
+  it "fetches provider versions form a custom registry secured by a token" do
+    hostname = "registry.example.org"
+    token = SecureRandom.hex(16)
+    credentials = [{ "type" => "terraform_registry", "host" => hostname, "token" => token }]
+
+    stub_request(:get, "https://#{hostname}/v1/providers/x/y/versions").
+      and_return(body: { "id": "x/y", "versions": [{ "version": "0.1.0" }] }.to_json)
+    client = described_class.new(hostname: hostname, credentials: credentials)
+
+    expect(client.all_provider_versions(identifier: "x/y")).to match_array([
+      Gem::Version.new("0.1.0")
+    ])
+    expect(WebMock).to have_requested(:get, "https://#{hostname}/v1/providers/x/y/versions").
+      with(headers: { "Authorization" => "Bearer #{token}" })
+  end
+
   it "fetches module versions", :vcr do
-    client = described_class.new(hostname: described_class::PUBLIC_HOSTNAME)
     response = client.all_module_versions(identifier: "hashicorp/consul/aws")
     expect(response.max).to eq(Gem::Version.new("0.9.3"))
   end
 
+  it "fetches module versions from a custom registry" do
+    hostname = "registry.example.org"
+    stub_request(:get, "https://#{hostname}/v1/modules/hashicorp/consul/aws/versions").
```

Comment:
Do we not VCR these because we don't have access to an actual private registry?

Reply:
> Do we not VCR these because we don't have access to an actual private registry?

Yes, that is one reason.